examples/meta/src/pipeline/pipeline.sg

```diff
@@ -13,7 +13,7 @@ Transformer pca = transformer("PCA")
 
 #![create_machine]
 Distance d = distance("EuclideanDistance", lhs=feats_train, rhs=feats_train)
-KMeans kmeans(distance=d, k=2)
+Machine kmeans = machine("KMeans", distance=d, k=2)
```

Comment:
why is this necessary to make it work?

Reply:
as in "what breaks otherwise?"

Reply:
I got this error from the CI https://dev.azure.com/shogunml/shogun/_build/results?buildId=279

Reply:
Mmmh the error doesn't make sense or? distance WAS provided....
Maybe this is worth investigating or would you say let's just move on? :D
Im on irc atm btw :)

Reply:
The generated code looks fine:
```
#![create_machine]
d = distance("EuclideanDistance", lhs=feats_train, rhs=feats_train)
kmeans = KMeans(distance=d, k=2)
#![create_machine]
```
Tried it out manually and get this for kmeans:
```python
In [11]: d = sg.distance("EuclideanDistance", lhs=feats_train, rhs=feats_train)
In [13]: kmeans = sg.KMeans(distance=d, k=2)
In [14]: kmeans
Out[14]: KMeans(cluster_centers={function},data_locked=false,dimensions=0,distance=null,k=3,labels=null,max_iter=10000,max_train_time=0,radiuses=Vector<double>(0): [],solver_type=0,store_model_features=true)
```
the issue is `distance=null`

Reply:
for reference, we discussed this in irc and it comes from using explicit ctors AND kwargs, which for now has to be avoided